source/agora/test/Ledger.d

```diff
@@ -36,6 +37,21 @@ private Block[] getAllBlocks (TestAPI node)
         if (new_blocks.length == 0)  // no blocks left
             break;
 
+        auto prev_height = new_blocks[0].header.height;
+
+        // consistency check: new blocks must continue from where we last called
+        // first loop: start with 0, subsequent: start from saved blocks position
+        if (blocks.length)
+            assert(prev_height == blocks.back.header.height + 1);
+        else
+            assert(prev_height == 0);  // genesis block
+
+        foreach (block; new_blocks[1 .. $])
+        {
+            // sequential consistency check
+            assert(block.header.height == ++prev_height);
```

Comment:
Yeah you could just reuse `starting_height` (maybe also rename it?) because those two have roughly the same function

Reply:
The second function verifies that the chain of blocks has an increasing height.

Reply:
So it's not like `[height0, height2, height1, height3]`.

Reply:
```d
size_t starting_block = 0;
while (1)
{
    auto new_blocks = node.getBlocksFrom(starting_block, uint.max);
    if (new_blocks.length == 0)  // no blocks left
        break;
    assert(new_blocks[0].header.height == starting_block)
    foreach (ref block; new_blocks[1 .. $])
        assert(block.header.height == ++starting_block);
    blocks ~= new_blocks;
}
```

What's wrong with this ?

Reply:
Nothing wrong, that's what reviews are for. Thanks!

Reply:
> What's wrong with this ?

I've solved the riddle. A missing `++starting_block` at the very end before the while loop ends, otherwise it's an infinite loop.